lib/server.go

```diff
@@ -817,6 +847,16 @@ func (srv *Server) _handleHeaderBundle(pp *Peer, msg *MsgDeSoHeaderBundle) {
 		len(msg.Headers), srv.blockchain.chainState(), pp,
 		srv.blockchain.headerTip().Header.Height, printHeight)))
 
+	// If the node is running a Fast-HotStuff validator and the consensus is running,
+	// in the steady-state, then it means that we don't sync the header chain separately.
+	if srv.fastHotStuffConsensus != nil && srv.fastHotStuffConsensus.IsRunning() {
```

Comment:
I will come back to this, but we used to use this in the steady-state to keep the node in sync as follows:
1. We call GetHeaders on a peer
2. They send us a header bundle
3. Based on that we may or may not fetch a block

If we're not doing the above, then we must have a different way of fetchin ga block when the consensus is running. Is this because blocks are "pushed" to us? I'll keep reading.

Reply:
Yes! We have a different and simpler way of fetching blocks once the new consensus is running:
1. Blocks are pushed to us without us having to request them in the new consensus
2. Whenever we see an orphan block with an unknown ancestor, we request the missing ancestor a peer until we have a fully connected chain